channels_redis/core.py

```diff
@@ -208,6 +215,9 @@ def _setup_encryption(self, symmetric_encryption_keys):
                     await self.receive_buffer[channel].put(message)
             else:
                 await self.receive_buffer[real_channel].put(message)
+            # this sleep makes sure that the redis queue is not popped before the
+            # previous message was processed and returned back to the caller
+            await asyncio.sleep(0.0001)
```

Comment:
Why does the sleep do this? Sleeping for 0.1 milliseconds sounds like a hack to ensure something else happens when in reality it should be a lock.

Reply:
well the sleep halts this coroutine after pushing the message in the channel receive buffer. so that the control moves to the receive coroutine before running the while loop again and it always does guarantee this, but anyways i am trying to come up with a better way to schedule the receive coroutine after a single pop from redis queue